From 7e4ff82fce72f590130dea72e8ace8339a51e954 Mon Sep 17 00:00:00 2001
From: Alejandro Sierra <b18039@freescale.com>
Date: Thu, 26 Nov 2015 05:25:09 -0600
Subject: [PATCH 06/21] Add CONFIG_SCMEVB to support diff on FreeX and EVB

CONFIG_SCMEVB to support differences between FreeX and EVB on the following:
- USDHC2 card detect pin
- Board name

Signed-off-by: Alejandro Sierra <b18039@freescale.com>
---
 board/freescale/mx6scmfreeX/mx6scmfreeX.c            | 16 ++++++++++++++--
 configs/mx6scm_1gb_evb_defconfig                     |  2 +-
 configs/mx6scm_1gb_evb_spinor_defconfig              |  2 +-
 configs/mx6scm_1gb_interleaving_evb_defconfig        |  2 +-
 configs/mx6scm_1gb_interleaving_evb_spinor_defconfig |  2 +-
 5 files changed, 18 insertions(+), 6 deletions(-)

diff --git a/board/freescale/mx6scmfreeX/mx6scmfreeX.c b/board/freescale/mx6scmfreeX/mx6scmfreeX.c
index 9a74f38..b6561cb 100644
--- a/board/freescale/mx6scmfreeX/mx6scmfreeX.c
+++ b/board/freescale/mx6scmfreeX/mx6scmfreeX.c
@@ -155,7 +155,11 @@ static iomux_v3_cfg_t const usdhc2_pads[] = {
 	MX6_PAD_NANDF_D5__SD2_DATA5	| MUX_PAD_CTRL(USDHC_PAD_CTRL),
 	MX6_PAD_NANDF_D6__SD2_DATA6	| MUX_PAD_CTRL(USDHC_PAD_CTRL),
 	MX6_PAD_NANDF_D7__SD2_DATA7	| MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_GPIO_4__GPIO1_IO04	| MUX_PAD_CTRL(NO_PAD_CTRL), /* CD */
+#ifdef CONFIG_SCMEVB
+	MX6_PAD_NANDF_D2__GPIO2_IO02    | MUX_PAD_CTRL(NO_PAD_CTRL), /* CD */
+#else
+	MX6_PAD_GPIO_4__GPIO1_IO04      | MUX_PAD_CTRL(NO_PAD_CTRL), /* CD */
+#endif
 };
 
 static iomux_v3_cfg_t const usdhc3_pads[] = {
@@ -360,8 +364,12 @@ int mmc_map_to_kernel_blk(int dev_no)
 {
 	return dev_no + 1;
 }
+#ifdef CONFIG_SCMEVB
+	#define USDHC2_CD_GPIO IMX_GPIO_NR(2, 2)
+#else
+	#define USDHC2_CD_GPIO  IMX_GPIO_NR(1, 4)
+#endif
 
-#define USDHC2_CD_GPIO	IMX_GPIO_NR(1, 4)
 #define USDHC3_CD_GPIO	IMX_GPIO_NR(2, 0)
 
 int board_mmc_getcd(struct mmc *mmc)
@@ -1215,7 +1223,11 @@ int board_late_init(void)
 
 int checkboard(void)
 {
+#ifdef CONFIG_SCMEVB
+	puts("Board: MX6SCM-EVB\n");
+#else
 	puts("Board: MX6SCM-FREEDOMX\n");
+#endif
 	return 0;
 }
 
diff --git a/configs/mx6scm_1gb_evb_defconfig b/configs/mx6scm_1gb_evb_defconfig
index f81c735..beed815 100644
--- a/configs/mx6scm_1gb_evb_defconfig
+++ b/configs/mx6scm_1gb_evb_defconfig
@@ -1,4 +1,4 @@
-CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\""
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",SCMEVB"
 CONFIG_ARM=y
 CONFIG_TARGET_MX6SCMFREEX=y
 CONFIG_DM=y
diff --git a/configs/mx6scm_1gb_evb_spinor_defconfig b/configs/mx6scm_1gb_evb_spinor_defconfig
index 2021ec2..ce38d78 100644
--- a/configs/mx6scm_1gb_evb_spinor_defconfig
+++ b/configs/mx6scm_1gb_evb_spinor_defconfig
@@ -1,4 +1,4 @@
-CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_BOOT_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\""
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_BOOT_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",SCMEVB"
 CONFIG_ARM=y
 CONFIG_TARGET_MX6SCMFREEX=y
 CONFIG_SYS_MALLOC_F=y
diff --git a/configs/mx6scm_1gb_interleaving_evb_defconfig b/configs/mx6scm_1gb_interleaving_evb_defconfig
index e9218cf..b7a8a84 100644
--- a/configs/mx6scm_1gb_interleaving_evb_defconfig
+++ b/configs/mx6scm_1gb_interleaving_evb_defconfig
@@ -1,4 +1,4 @@
-CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",INTERLEAVING_MODE"
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",INTERLEAVING_MODE,SCMEVB"
 CONFIG_ARM=y
 CONFIG_TARGET_MX6SCMFREEX=y
 CONFIG_DM=y
diff --git a/configs/mx6scm_1gb_interleaving_evb_spinor_defconfig b/configs/mx6scm_1gb_interleaving_evb_spinor_defconfig
index e29a927..e7908e1 100644
--- a/configs/mx6scm_1gb_interleaving_evb_spinor_defconfig
+++ b/configs/mx6scm_1gb_interleaving_evb_spinor_defconfig
@@ -1,4 +1,4 @@
-CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_BOOT_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",INTERLEAVING_MODE"
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6scmfreeX/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_BOOT_SPINOR,DEFAULT_FDT_FILE=\"imx6scm-1gb-evb-ldo.dtb\",INTERLEAVING_MODE,SCMEVB"
 CONFIG_ARM=y
 CONFIG_TARGET_MX6SCMFREEX=y
 CONFIG_SYS_MALLOC_F=y
-- 
1.9.1

